docs/contributors/guidelines.rst

```diff
@@ -0,0 +1,205 @@
+constraints, some are architectural opinions.
+
+If you contribute to this project, please read through them all and read the
+architectural documentation. Reviewers will check that your contributions
+match these guidelines amont other things.
+
+If you feel one of the guidelines is wrong or incomplete, please come and
+discuss on our messaging channel, then open a pull request including the
+change you would like to make and the reasons why in the description, we can
+then formally discuss it and review it following our usual review process.
+
+General
+-------
+
+What features should be included
+````````````````````````````````
+
+Mailu is a mail server. That said, mail works in an ecosystem and we should set
+clear boundaries on what Mailu is and what it is not. Beyond mail, we include
+features that mail is dependent on in some way, not the other way aroud.
+
+For instance, we include contact management and a dav service for synchronizing
+contacts, because sending mail is made easier with manageable contacts. We do
+not include a calendar management app or file sharing app, because mail is not
+dependent on them.
+
+Examples of features we will not include in Mailu are: calendars, documents and
+file sharing, full-sized groupware, instant messaging, password management. If
+you want and success in connecting those to Mailu and want to share, please
+write some useful documentation for others to do the same.
+
+What behavior is tolerated
+``````````````````````````
+
+Be respectful of others and others contributions. If you find a whole part of
+Maiu is terribly written or badly designed, please open an issue to discuss it
+calmly or come and have a chat on our channel.
+
+Any PR that just rewrites a huge part of the code by making mostly cosmetic or
+opinionated changes without prior discussion will be received with a polite
+"no".
+
+Architecture
+------------
+
+What setups should be supported
+```````````````````````````````
+
+Mailu supports out-of-the-box Docker Compose, Docker Swarm and Kubernetes
+environments. In those environments, it consists of many containers and
+supports hosting some of those containers in a separate environment.
+
+It supports hosting the data outside of the environment, in as many types of
+database servers as can be documented.
+
+It supports spreading and scaling the containers across hosts, except for
+those that share volumes. The intention is to stop sharing volumes between
+containers.
+
+What is a Mailu container
+`````````````````````````
+
+A Mailu container should provide one service and run one type of process only.
+A new Webmail should be in a separate container, a new antivirus or a new
+antispam should be in a separate container.
+
+A container is developped as a singled directory under the proper category in
+the main repository, the only exception being service containers that should
+only use official Docker images.
```

Comment:
Yes I mean redis, maybe later some others (no specific plans, but heh, could happen). Maybe utility containers would avoid confusion?

Reply:
I'm thinking of the current state of the `services` directory in the repo:
- Fetchmail -> Is actually `optional`
- Rspamd -> Postfix breaks without it, perhaps should be `core`?
- Unbound -> Is actually `optional`

So perhaps we can define 3 groups, instead of 4:
1. Core
2. Optional
3. Webmails

What's your view on this?

Reply:
Agreed with the 3-directory approach :)

Reply:
In 12factor.net , they use the term backing service.

Maybe you can rephrase it and call them stock backing service as opposed to customized.

Stock would be redis because you consume the image as is.
And customized, would be postfix as you highly customize it (in the event there would be an official postfix docker image)